### DISCUSSION TITLE
How to get "first" item from filtered element?

### DISCUSSION THREAD

**Discussion by quad5** [cypress-io/cypress#15413] [Questions and Help] [answered]
It seems like `first` API get first DOM element period, not first of filtered set.

For example: `cy.get('.class1').not('.class2').first().then((element) => {}
)`. 

Example above will get first of `.class1` and ignore `.class2`.

**Comment by bahmutov** [comment, marked as answer]
Can you provide a reproducible example? For now, we have the following example https://glebbahmutov.com/cypress-examples/6.6.0/recipes/find-by-class.html#filter-elements-using-cy-not showing the test grabbing the correct first filtered item

```html
<ul>
  <li class="initial active">Apples</li>
  <li class="initial current">Oranges</li>
  <li class="initial active">Grapes</li>
  <li class="initial current">Melons</li>
</ul>
```

```js
cy.get('.initial')
  .not('.active')
  .should('have.length', 2)
  .first()
  .should('have.text', 'Oranges')
```

```

**Comment by quad5** [comment]
Hmm..strange..I am not seeing it now. Will update when I do find it.
